Lib/test/test_copy.py

```diff
@@ -936,14 +936,22 @@ def __replace__(self, **changes):
 
     def test_namedtuple(self):
         from collections import namedtuple
-        Point = namedtuple('Point', 'x y', defaults=(0,))
-        p = Point(11, 22)
-        self.assertEqual(copy.replace(p), (11, 22))
-        self.assertEqual(copy.replace(p, x=1), (1, 22))
-        self.assertEqual(copy.replace(p, y=2), (11, 2))
-        self.assertEqual(copy.replace(p, x=1, y=2), (1, 2))
-        with self.assertRaisesRegex(ValueError, 'unexpected field name'):
-            copy.replace(p, x=1, error=2)
+        from typing import NamedTuple
+        PointFromCall = namedtuple('Point', 'x y', defaults=(0,))
+        class PointFromInheritance(PointFromCall):
+            pass
+        class PointFromClass(NamedTuple):
+            x: int
+            y: int = 0
+        for Point in (PointFromCall, PointFromInheritance, PointFromClass):
+            with self.subTest(Point=Point):
+                p = Point(11, 22)
```

Comment:
Should we test the class?

```suggestion
                p = Point(11, 22)
                assert isinstance(copy.replace(p, x=12), Point)
```

Reply:
Why not? :)
Done.

Reply:
Oh I meant to test the copy, not the original p!